execution-engine/src/wasi/wasmi.rs

```diff
@@ -40,25 +40,60 @@ impl HostError for FatalEngineError {}
 /// Impl the MemoryHandler for MemoryRef.
 /// This allows passing the MemoryRef to WasiWrapper on any VFS call.
 impl MemoryHandler for MemoryRef {
-    /// Writes a buffer of bytes, `buffer`, to the runtime state's memory at
-    /// address, `address`.  Success on returing ErrNo::Success,
-    /// Fails with `ErrNo::NoMem` if no memory is registered in the runtime state.
-    #[inline]
-    fn write_buffer(&mut self, address: u32, buffer: &[u8]) -> FileSystemResult<()> {
-        if let Err(_e) = self.set(address, buffer) {
-            return Err(ErrNo::NoMem);
-        }
-        Ok(())
+    type Slice = &'static [u8];
+    type SliceMut = &'static mut [u8];
+
+    fn get_slice<'a>(
+        &'a self,
+        address: u32,
+        length: u32
+    ) -> FileSystemResult<Bound<'a, Self::Slice>> {
+        let address = usize::try_from(address).unwrap();
```

Comment:
unwrap can be converted to ErrNo::Inval.

Reply:
I'll run a sed command on all of these